File: 1.0/src/cz/mg/vulkan/PFNvkGetPhysicalDeviceFormatProperties2.java
package cz.mg.vulkan;

public class PFNvkGetPhysicalDeviceFormatProperties2 extends VkFunctionPointer {
    public PFNvkGetPhysicalDeviceFormatProperties2() {
    }

    protected PFNvkGetPhysicalDeviceFormatProperties2(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetPhysicalDeviceFormatProperties2(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetPhysicalDeviceFormatProperties2(long value) {
        setValue(value);
    }

    public PFNvkGetPhysicalDeviceFormatProperties2(VkInstance instance) {
        super(instance, new VkString("vkGetPhysicalDeviceFormatProperties2"));
    }

    public void call(VkPhysicalDevice physicalDevice, int format, VkFormatProperties2 pFormatProperties){
        callNative(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.getNullAddressNative(), format, pFormatProperties != null ? pFormatProperties.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long physicalDevice, int format, long pFormatProperties);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetPhysicalDeviceFormatProperties2.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetPhysicalDeviceFormatProperties2_callNative(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jint format, jlong pFormatProperties){
    (void)env;
    (void)jc;
    PFN_vkGetPhysicalDeviceFormatProperties2 f = (PFN_vkGetPhysicalDeviceFormatProperties2)jniLongToFunctionPointer(address);
    f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        (VkFormat)format,
        ((VkFormatProperties2*)jniLongToPointer(pFormatProperties))
    );
}

